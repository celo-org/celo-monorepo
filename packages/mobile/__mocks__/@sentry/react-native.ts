const Sentry = {
  config: () => ({ install: jest.fn() }),
  setTagsContext: jest.fn(),
  captureException: jest.fn(),
}

module.exports = Sentry
