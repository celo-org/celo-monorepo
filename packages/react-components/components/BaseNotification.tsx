import TextButton from '@celo/react-components/components/TextButton'
import Touchable from '@celo/react-components/components/Touchable'
import colors from '@celo/react-components/styles/colors'
import fontStyles from '@celo/react-components/styles/fonts'
import { elevationShadowStyle } from '@celo/react-components/styles/styles'
import variables from '@celo/react-components/styles/variables'
import * as React from 'react'
import { StyleSheet, Text, View } from 'react-native'

const { contentPadding } = variables

interface Props {
  icon?: React.ReactNode
  title: string
  children: React.ReactNode
  ctas: CTA[]
  onPress?: () => unknown
}

export interface CTA {
  onPress: () => unknown
  text: string
}

function Wrapper({ onPress, children }: { children: React.ReactNode; onPress?: () => unknown }) {
  return onPress ? (
    <Touchable style={styles.wrapper} onPress={onPress}>
      {children}
    </Touchable>
  ) : (
    <View style={styles.wrapper}>{children}</View>
  )
}

export default function BaseNotification({ icon, title, children, ctas, onPress }: Props) {
  return (
    <View style={[styles.container, elevationShadowStyle(2)]}>
      <Wrapper onPress={onPress}>
        <View style={styles.innerContainer}>
          {icon && <View style={styles.iconArea}>{icon}</View>}
          <View style={styles.contentArea}>
            <Text style={fontStyles.bodySmallSecondary}>{title}</Text>
            <View style={styles.body}>
              {children}
              <View style={styles.ctas}>
                {ctas.map((cta, j) => (
                  <TextButton key={j} style={styles.action} onPress={cta.onPress}>
                    {cta.text}
                  </TextButton>
                ))}
              </View>
            </View>
          </View>
        </View>
      </Wrapper>
    </View>
  )
}

const styles = StyleSheet.create({
  ctas: {
    flexDirection: 'row',
    marginTop: 5,
  },
  action: {
    paddingEnd: 15,
  },
  body: {
    paddingTop: 8,
    minHeight: 60,
    justifyContent: 'space-between',
  },
  container: {
    width: '100%',
    backgroundColor: colors.background,
  },
  innerContainer: {
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  wrapper: {
    padding: contentPadding,
  },
  iconArea: {
    paddingRight: contentPadding,
    alignItems: 'center',
  },
  contentArea: {
    justifyContent: 'space-between',
    flex: 1,
  },
})
