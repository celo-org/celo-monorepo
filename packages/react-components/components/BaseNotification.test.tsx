import BaseNotification from '@celo/react-components/components/BaseNotification'
import { shallow } from 'enzyme'
import * as React from 'react'
import { Text } from 'react-native'
import * as renderer from 'react-test-renderer'

const props = (onPress = jest.fn()) => ({
  text: 'Gold is where you can choose to store Celo dollars you have',
  image: '',
  title: 'Test',
  ctas: [{ text: 'it goes boom', onPress }],
})

describe(BaseNotification, () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <BaseNotification {...props()}>
        <Text>Test</Text>
      </BaseNotification>
    )
    expect(tree).toMatchSnapshot()
  })
  describe('when ctas are pressed', () => {
    it('calls the on press function', () => {
      const clickHandler = jest.fn()

      const wrapper = shallow(
        <BaseNotification {...props(clickHandler)}>
          <Text>Test</Text>
        </BaseNotification>
      )
      wrapper
        .find('TextButton')
        .first()
        .simulate('press')

      expect(clickHandler).toHaveBeenCalled()
    })
  })
})
