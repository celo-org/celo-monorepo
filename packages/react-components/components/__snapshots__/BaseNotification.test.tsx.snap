// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseNotification renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#FFFFFF",
        "width": "100%",
      },
      Object {
        "elevation": 2,
        "shadowColor": "black",
        "shadowOffset": Object {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.3,
        "shadowRadius": 1.6,
      },
    ]
  }
>
  <View
    style={
      Object {
        "padding": 16,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#81868B",
              "fontFamily": "Hind-Regular",
              "fontSize": 14,
              "lineHeight": 18,
            }
          }
        >
          Test
        </Text>
        <View
          style={
            Object {
              "justifyContent": "space-between",
              "minHeight": 60,
              "paddingTop": 8,
            }
          }
        >
          <Text>
            Test
          </Text>
          <View
            style={
              Object {
                "flexDirection": "row",
                "marginTop": 5,
              }
            }
          >
            <View
              accessible={true}
              focusable={true}
              isTVSelectable={true}
              nativeBackgroundAndroid={
                Object {
                  "attribute": "selectableItemBackgroundBorderless",
                  "type": "ThemeAttrAndroid",
                }
              }
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#42D689",
                      "fontFamily": "Hind-SemiBold",
                      "fontSize": 14,
                      "lineHeight": 18,
                    },
                    Object {
                      "paddingEnd": 15,
                    },
                  ]
                }
              >
                it goes boom
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
